docs/components/DocumentationPage.tsx

```diff
@@ -117,7 +117,10 @@ export default function DocumentationPage(props: Props) {
       isMobileMenuVisible={isMobileMenuVisible}
       onContentScroll={handleContentScroll}
       sidebarScrollPosition={sidebarScrollPosition}>
-      <Head title={props.title} description={props.description}>
+      <Head
+        title={props.title}
+        description={props.description}
+        canonicalUrl={version !== 'unversioned' ? getCanonicalUrl(pathname) : undefined}>
```

Comment:
@simek, I removed the `!version` check since we still want the canonical URL for the versioned pages to point to the latest URL.

Here are is a  screenshot when running docs locally for example for SDK 47:

![CleanShot 2023-09-26 at 23 02 59](https://github.com/expo/expo/assets/10234615/2db99c5f-bd78-4228-bff2-b8b704515bec)

For the latest version which was missing the canonical URL before (on production - it will be a self referencing canonical URL)

![CleanShot 2023-09-26 at 23 03 16](https://github.com/expo/expo/assets/10234615/13a107ba-69f5-4207-80eb-a7581d72adf1)